ocs_ci/ocs/cluster.py

```diff
@@ -1321,7 +1323,110 @@ def get_ceph_df_detail():
     """
     ceph_cmd = "ceph df detail"
     ct_pod = pod.get_ceph_tools_pod()
-    return ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd, format="json-pretty")
+    return ct_pod.exec_ceph_cmd(
+        ceph_cmd=ceph_cmd, format=format, out_yaml_format=out_yaml_format
+    )
+
+
+def parse_ceph_df_pools(raw_output: str) -> pd.DataFrame:
+    """
+    Parse the 'ceph df detail' command output and extract the POOLS section into a pandas DataFrame.
+
+    Args:
+        raw_output (str): The raw output string from the 'ceph df detail' command.
+
+    Returns:
+        pd.DataFrame: A pandas DataFrame containing the parsed POOLS section data.
+            The DataFrame includes columns for POOL, ID, PGS, STORED, OBJECTS, USED,
+            %USED, MAX AVAIL, QUOTA OBJECTS, QUOTA BYTES, DIRTY, USED COMPR, and UNDER COMPR.
+
+    Note:
+        This function assumes a specific format for the 'ceph df detail' output.
+        It extracts the POOLS section, processes the header and data rows,
+        and returns a structured DataFrame for further analysis.
+
+    """
+    pools_section = (
+        re.search(r"--- POOLS ---\n(.*)", raw_output, re.DOTALL).group(1).strip()
+    )
+    pools_lines = [line.strip() for line in pools_section.split("\n") if line.strip()]
+    header = [
+        "POOL",
+        "ID",
+        "PGS",
+        "STORED",
+        "(DATA)",
+        "(OMAP)",
+        "OBJECTS",
+        "USED",
+        "(DATA)",
+        "(OMAP)",
+        "%USED",
+        "MAX AVAIL",
+        "QUOTA OBJECTS",
+        "QUOTA BYTES",
+        "DIRTY",
+        "USED COMPR",
+        "UNDER COMPR",
+    ]
+    logger.info(f"Number of columns: {len(header)}")
+    data = []
+    for line in pools_lines[1:]:
+        parts = re.split(r"\s{2,}", line.strip())
+        if len(parts) == len(header):
+            data.append(parts)
+        else:
+            logger.warning(f"Mismatch in column count for line: {line}")
+            logger.warning(f"Expected {len(header)} columns, got {len(parts)}")
+    df = pd.DataFrame(data, columns=header)
+
+    return df
+
+
+def ceph_details_df_to_dict(df: pd.DataFrame) -> dict:
+    """
+    Convert the DataFrame to a dictionary where the POOL column is the key
+    and the rest of the columns form a nested dictionary.
+
+    Args:
+        df (pd.DataFrame): A pandas DataFrame containing Ceph pool information.
+
+    Returns:
+        dict: A dictionary where each key is a pool name, and the corresponding value
+              is a nested dictionary containing the rest of the columns' data for that pool.
+
+    """
+    return {row["POOL"]: row.drop("POOL").to_dict() for _, row in df.iterrows()}
+
+
+def validate_num_of_pgs(expected_pgs: dict) -> bool:
```

Comment:
this is example for usage, i will update doc string so it will be clear. it can check one pool or few
```
expected_pgs = {
            "ocs-storagecluster-cephblockpool": 1,
            "ocs-storagecluster-cephfilesystem-data0": 1,
            "ocs-storagecluster-cephfilesystem-metadata0": 1,
        }
        ```

Reply:
discussed offline